forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -844,7 +844,7 @@ public boolean turnFaceUp(boolean runTriggers, SpellAbility cause) {
             game.getReplacementHandler().run(ReplacementType.TurnFaceUp, repParams);
 
             // Run triggers
-            final Map<AbilityKey, Object> runParams = AbilityKey.mapFromCard(this);
+            final Map<AbilityKey, Object> runParams = AbilityKey.mapFromCard(CardCopyService.getLKICopy(this));
```

Comment:
like I said this does not really fix anything
this trigger should not look back in time by default
now when casting _Giant Growth_ on it in response won't increase the damage it deals...

Reply:
The Problem is related to https://github.com/Card-Forge/forge/pull/117

DamageDealEffect needs to be updated to handle LKI info "IF" needed

we have updated other effects to be able to use the GameObject while the Card is still in the same zone,
but fall back to LKI when it isn't

Reply:
@tool4ever the bigger Problem:

DamageSource might be able to handle LKI and update it to GameObject
calculateAmount, the function that gets the Number with `TriggeredCard$CardPower` is not able to do that

Reply:
DamageDealEffect should already convert its source into LKI if able (from changeZoneLKI)

but there is no logic for converting it back, so this would also use source from wrong time